assignments/serializers.py

```diff
@@ -114,3 +116,70 @@ def save(self):
             student=student_instance,
             **budgeting_text_data) for budgeting_text_data in self.validated_data['budgeting_targets']]
         BudgetingTargetAnswer.objects.bulk_create(budgeting_target_answers)
+
+
+# serializers used for getting answers for report generation
+class ReportOpenTextTaskSerializer(serializers.ModelSerializer):
+    answers = serializers.SerializerMethodField()
+
+    class Meta:
+        model = OpenTextTask
+        fields = ['question', 'answers']
+
+    def get_answers(self, obj):
+        answers = obj.open_text_answers.filter(**self.context['query_params'])
+        serializer = OpenTextAnswerSerializer(instance=answers, many=True)
+        return serializer.data
+
+
+class ReportBudgetingTargetSerializer(serializers.ModelSerializer):
+    target = BudgetingTargetSerializer()
+
+    class Meta:
+        model = BudgetingTargetAnswer
+        fields = ['amount', 'point', 'target']
+
+
+class ReportBudgetingTaskSerializer(serializers.ModelSerializer):
+    targets = serializers.SerializerMethodField()
+
+    class Meta:
+        model = BudgetingTask
+        fields = ['name', 'targets']
+
+    def get_targets(self, obj):
+        targets = obj.budgeting_answers.filter(**self.context['query_params'])
+        serializer = ReportBudgetingTargetSerializer(instance=targets, many=True)
+        return serializer.data
+
+
+class ReportSectionSerializer(serializers.ModelSerializer):
+    open_text_tasks = ReportOpenTextTaskSerializer(many=True, source='opentexttasks')
+    budgeting_tasks = ReportBudgetingTaskSerializer(many=True, source='budgetingtasks')
+
+    class Meta:
+        model = Section
+        fields = ['title', 'open_text_tasks', 'budgeting_tasks']
+
+
+class ReportAssignmentSerializer(serializers.ModelSerializer):
+    students = serializers.SerializerMethodField()
+    sections = ReportSectionSerializer(many=True)
+
+    class Meta:
+        model = Assignment
+        fields = ['name', 'sections', 'students']
+
+    def get_students(self, obj):
+        student_query = dict(
+            [(key.replace('student__', ''), value) for key, value in self.context['query_params'].items()])
+        students_per_school = Student.objects.filter(
+            school__assignment__slug=self.context['assignment_slug'],
+            **student_query).values('school__name').annotate(count=Count('id')).order_by()
```

Comment:
So the `**student_query` is used to filter answers for specific school and class?

This whole method feels a bit complicated. Could you move some of the logic into `Assignment` and try to have something like:
```
answers_per_school = Assignment.get_answers(school=school).count()
answers_per_school = Assignment.get_answers(school_class=school_class).count()
```
Then in `Assignment` model's `get_answers` you could encapsulate the more complex querying. Having such methods in `Assignment` makes the code also more reusable than having complex logic in specific serializer.